components/raftstore-v2/src/raft/peer.rs

```diff
@@ -792,4 +811,57 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
     pub fn gc_peer_context_mut(&mut self) -> &mut GcPeerContext {
         &mut self.gc_peer_context
     }
+
+    #[inline]
+    pub fn update_last_sent_snapshot_index(&mut self, i: u64) {
+        if i > self.last_sent_snapshot_index {
+            self.last_sent_snapshot_index = i;
+        }
+    }
+
+    #[inline]
+    pub fn last_sent_snapshot_index(&self) -> u64 {
+        self.last_sent_snapshot_index
+    }
+
+    /// Returns (minimal matched, minimal committed)
+    pub fn get_min_progress(&self) -> Result<(u64, u64)> {
```

Comment:
Should be in merge module as it's only used for merge. And should be named as `min_progress`.